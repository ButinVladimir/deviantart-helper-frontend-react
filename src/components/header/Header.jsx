import React from 'react';
import PropTypes from 'prop-types';
import Container from 'react-bulma-components/lib/components/container';
import Navbar from 'react-bulma-components/lib/components/navbar';
import HeaderNotLoggedIn from './HeaderNotLoggedInContainer';
import HeaderLoggedIn from './HeaderLoggedInContainer';

export default function Header({
  isLoggedIn,
  menuToggled,
}) {
  return (
    <Navbar color="dark" active={menuToggled}>
      <Container breakpoint="fullhd">
        {!isLoggedIn && <HeaderNotLoggedIn />}
        {isLoggedIn && <HeaderLoggedIn />}
      </Container>
    </Navbar>
  );
}

Header.propTypes = {
  isLoggedIn: PropTypes.bool.isRequired,
  menuToggled: PropTypes.bool.isRequired,
};
