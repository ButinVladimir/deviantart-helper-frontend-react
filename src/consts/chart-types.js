export const CHART_VIEWS = 'views';
export const CHART_FAVOURITES = 'favourites';
export const CHART_COMMENTS = 'comments';
export const CHART_DOWNLOADS = 'downloads';
