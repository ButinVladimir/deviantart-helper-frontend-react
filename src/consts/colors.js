const LIGHT_BLUE = '#88f';
const RED = '#b33';
const GREEN = '#393';
const BLUE = '#11c';
const ORANGE = '#f94';
const PURPLE = '#94a';
const CYAN = '#2b9';
const BLACK = '#111';
const GREY = '#999';
const PINK = '#d5b';

export default [LIGHT_BLUE, RED, GREEN, BLUE, ORANGE, PURPLE, CYAN, BLACK, GREY, PINK];
