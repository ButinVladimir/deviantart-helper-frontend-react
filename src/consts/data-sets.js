export const DATA_SET_TIMESTAMP = 0;
export const DATA_SET_VIEWS = 1;
export const DATA_SET_FAVOURITES = 2;
export const DATA_SET_COMMENTS = 3;
export const DATA_SET_DOWNLOADS = 4;

export const dataSetOptions = [
  [DATA_SET_VIEWS, 'Views'],
  [DATA_SET_FAVOURITES, 'Favourites'],
  [DATA_SET_COMMENTS, 'Comments'],
  [DATA_SET_DOWNLOADS, 'Downloads'],
];
