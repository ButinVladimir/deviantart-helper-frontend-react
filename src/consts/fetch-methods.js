export const GET = 'GET';
export const POST = 'POST';
