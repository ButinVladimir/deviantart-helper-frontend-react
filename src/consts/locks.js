export const LOCK_LOAD_USER_INFO = 'LOCK_LOAD_USER_INFO';
export const LOCK_START_LOADING_DATA = 'LOCK_START_LOADING_DATA';
export const LOCK_REVOKE = 'LOCK_REVOKE';

export const LOCK_BROWSE_DEVIATIONS = 'LOCK_BROWSE_DEVIATIONS';
export const LOCK_DEVIATION_DETAILS = 'LOCK_DEVIATION_DETAILS';
export const LOCK_DEVIATION_DETAILS_METADATA = 'LOCK_DEVIATION_DETAILS_METADATA';
export const LOCK_DEVIATIONS_STATISTICS = 'LOCK_DEVIATIONS_STATISTICS';
export const LOCK_DEVIATIONS_TOTAL = 'LOCK_DEVIATIONS_TOTAL';
