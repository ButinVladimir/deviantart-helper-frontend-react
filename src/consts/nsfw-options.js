export const SHOW_ALL = 'SHOW_ALL';
export const SHOW_NON_NSFW = 'SHOW_NON_NSFW';
export const SHOW_NSFW = 'SHOW_NSFW';

export const nsfwOptions = [
  [SHOW_ALL, 'Show everything'],
  [SHOW_NON_NSFW, 'Show non-NSFW'],
  [SHOW_NSFW, 'Show NSFW'],
];
