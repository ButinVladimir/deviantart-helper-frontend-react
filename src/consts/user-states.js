export const NOT_LOGGINED = 0;
export const FETCHING_INFO = 1;
export const FULLY_LOGINNED = 2;
