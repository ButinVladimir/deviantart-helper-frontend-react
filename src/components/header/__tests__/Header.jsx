import React from 'react';
import { shallow } from 'enzyme';
import Header from '../Header';
import * as userStates from '../../../consts/user-states';

describe('Message', () => {
  it('renders correctly when user state is not loginned', () => {
    const wrapper = shallow(
      <Header
        userState={userStates.NOT_LOGGINED}
        menuToggled={false}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly when user state is fetching info', () => {
    const wrapper = shallow(
      <Header
        userState={userStates.FETCHING_INFO}
        menuToggled={false}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly when user state is fully loginned', () => {
    const wrapper = shallow(
      <Header
        userState={userStates.FULLY_LOGINNED}
        menuToggled
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
