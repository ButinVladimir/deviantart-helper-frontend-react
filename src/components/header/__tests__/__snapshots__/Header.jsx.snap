// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message renders correctly when user state is fetching info 1`] = `
<Navbar
  active={false}
  className=""
  color="dark"
  fixed={null}
  renderAs="nav"
  style={Object {}}
  transparent={false}
>
  <Container
    breakpoint="fullhd"
    fluid={false}
    renderAs="div"
    style={Object {}}
  >
    <Component />
  </Container>
</Navbar>
`;

exports[`Message renders correctly when user state is fully loginned 1`] = `
<Navbar
  active={true}
  className=""
  color="dark"
  fixed={null}
  renderAs="nav"
  style={Object {}}
  transparent={false}
>
  <Container
    breakpoint="fullhd"
    fluid={false}
    renderAs="div"
    style={Object {}}
  >
    <Component />
  </Container>
</Navbar>
`;

exports[`Message renders correctly when user state is not loginned 1`] = `
<Navbar
  active={false}
  className=""
  color="dark"
  fixed={null}
  renderAs="nav"
  style={Object {}}
  transparent={false}
>
  <Container
    breakpoint="fullhd"
    fluid={false}
    renderAs="div"
    style={Object {}}
  >
    <Component />
  </Container>
</Navbar>
`;
